baybe/surrogates/custom.py

```diff
@@ -23,97 +23,27 @@
-    batchify_posterior: bool = True,
-) -> Callable:
-    """Wrap a given custom model architecture class into a ```Surrogate```.
-
-    Args:
-        joint_posterior_attr: Boolean indicating if the model returns a posterior
-            distribution jointly across candidates or on individual points.
-        constant_target_catching: Boolean indicating if the model cannot handle
-            constant target values and needs the @catch_constant_targets decorator.
-        batchify_posterior: Boolean indicating if the model is incompatible
-            with t- and q-batching and needs the @batchify decorator for its posterior.
-
-    Returns:
-        A function that wraps around a model class based on the specifications.
-    """
-
-    def construct_custom_architecture(model_cls):
-        """Construct a surrogate class wrapped around the custom class."""
-        validate_custom_architecture_cls(model_cls)
-
-        class CustomArchitectureSurrogate(Surrogate):
-            """Wraps around a custom architecture class."""
-
-            joint_posterior: ClassVar[bool] = joint_posterior_attr
-            supports_transfer_learning: ClassVar[bool] = False
-
-            def __init__(self, *args, **kwargs):
-                self._model = model_cls(*args, **kwargs)
-
-            def _fit(
-                self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
-            ) -> None:
-                return self._model._fit(searchspace, train_x, train_y)
-
-            def _posterior(self, candidates: Tensor) -> tuple[Tensor, Tensor]:
-                return self._model._posterior(candidates)
-
-            def __get_attribute__(self, attr):
-                """Access the attributes of the class instance if available.
-
-                If the attributes are not available,
-                it uses the attributes of the internal model instance.
-                """
-                # Try to retrieve the attribute in the class
-                try:
-                    val = super().__getattribute__(attr)
-                except AttributeError:
-                    pass
-                else:
-                    return val
-
-                # If the attribute is not overwritten, use that of the internal model
-                return self._model.__getattribute__(attr)
-
-        # Catch constant targets if needed
-        cls = (
-            catch_constant_targets(CustomArchitectureSurrogate)
-            if constant_target_catching
-            else CustomArchitectureSurrogate
-        )
-
-        # batchify posterior if needed
-        if batchify_posterior:
-            cls._posterior = batchify(cls._posterior)
-
-        # Block serialization of custom architectures
-        converter.register_unstructure_hook(
-            CustomArchitectureSurrogate, block_serialization_hook
-        )
-
-        return cls
-
-    return construct_custom_architecture
+def register_custom_architecture(*args, **kwargs) -> NoReturn:
```

Comment:
I think in the past there was simply no case where it was required? If so, we probably overlooked and wrote `None` by mistake. Do you see one?

Reply:
`NoReturn` seems correct here as no return
makes me think tho, why is it not required to mark functions that have a possible raise as `-> normal_return_type | NoReturn` ?

Reply:
Actually, good point, but I think the emphasis here is on "never"
<img width="623" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/97deb594-f43e-4029-84f0-154e16d56eb7">
Because otherwise, you'd basically need to add this **everywhere** because any function can forward an exception from a lower level (even the function itself has no raise statement), and that would defeat the purpose